src/pluginHost.node/sandbox/env.ts

```diff
@@ -0,0 +1,33 @@
+import { env as originalEnv } from 'node:process'
+
+// prettier-ignore
+const allowList = [
+	'BLOCKSIZE',
+	'COLORFGBG', 'COLORTERM',
+	'CHARSET', 'LANG', 'LANGUAGE',
+	'LC_ALL', 'LC_COLLATE', 'LC_CTYPE',
+	'LC_MESSAGES', 'LC_MONETARY', 'LC_NUMERIC', 'LC_TIME',
+	'LINES COLUMNS',
+	'LSCOLORS',
+	'SSH_AUTH_SOCK',
+	'TZ',
+	'DISPLAY', 'XAUTHORIZATION', 'XAUTHORITY',
+	'EDITOR', 'VISUAL',
+	'HOME', 'MAIL', 'PATH',
+	// Linux
+	'DBUS_SESSION_BUS_ADDRESS',
+	'XAPPLRESDIR', 'XFILESEARCHPATH', 'XUSERFILESEARCHPATH',
+	'QTDIR', 'KDEDIR',
+	'XDG_SESSION_COOKIE',
+	'XMODIFIERS', 'GTK_IM_MODULE', 'QT_IM_MODULE', 'QT_IM_SWITCHER',
+	// Windows
+	'APPDATA', 'LOCALAPPDATA',
+	'ProgramData', 'ProgramFiles',
+	'SystemDrive', 'SystemRoot',
+]
+
+export const permittedEnv = Object.fromEntries(
+	Object.entries(originalEnv)
+		.filter(([ name ]) => allowList.includes(name))
+		.filter(([ name ]) => name !== 'ELECTRON_RUN_AS_NODE')
```

Comment:
thought: we don't need this? `ELECTRON_RUN_AS_NODE` is not in the allowlist. We should add the comment back to provide contexts.

```text
this will also unset ELECTRON_RUN_AS_NODE so that the child process can use node/electron based applications
```

Reply:
We should remove any envs inserted by Electron to get original envs before launching Electron.

Reply:
fixed